### PR TITLE
Proposal: Donate the MPI-Operator.V2 to kubernetes-sigs

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
https://github.com/kubeflow/mpi-operator/issues/459

**Comment by terrytangyuan** [comment]
@kubeflow/project-steering-group

File: proposals/donate-mpi-operator.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+* 2022-03-11 - Proposed 
+
+## Motivation
+Kubeflow currently is moving to an [Unified operator](https://github.com/kubeflow/training-operator)
+The motivation is to encourage non-training users (like HPC) to use and contribute to it, without having to install or learn about kubeflow's training-operator.
+With the creation of the project [k8s-sigs/Kueue](https://github.com/kubernetes-sigs/kueue), having the MPi-Operator as a k8s-sigs project will facilitate the efforts to create and maintain a custom job queueing mechasinm for mpi jobs on kubernetes.
+
+## Goals
+* Migrate repo kubeflow/mpi-operatorv2 to kubernetes-sigs/mpi-operator
```
Comment (alculquicondor) [inline proposals/donate-mpi-operator.md:24]:
```suggestion
* Migrate repo kubeflow/mpi-operator/v2 to kubernetes-sigs/mpi-operator
```

?

File: proposals/donate-mpi-operator.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+_Status_
+
+* 2022-03-11 - Proposed 
+
+## Motivation
+Kubeflow currently is moving to an [Unified operator](https://github.com/kubeflow/training-operator)
+The motivation is to encourage non-training users (like HPC) to use and contribute to it, without having to install or learn about kubeflow's training-operator.
+With the creation of the project [k8s-sigs/Kueue](https://github.com/kubernetes-sigs/kueue), having the MPi-Operator as a k8s-sigs project will facilitate the efforts to create and maintain a custom job queueing mechasinm for mpi jobs on kubernetes.
```
[the remark this answers is not in the document]
Comment (ArangoGutierrez) [reply proposals/donate-mpi-operator.md:21]: will rm

**Comment by theadactyl** [comment]
Thanks for opening this! I don't see anywhere an assessment of how this impacts existing users of MPI operator and/or the unified operator, and what the implications are for unified operator going forward. Also how this might impact other parts of Kubeflow. I think that's important to include -- can you do so?

Additionally, what in particular is the value of donation here, vs just maintaining MPI operator within Kubeflow with fewer dependencies that would block HPC users? Donation isn't a lightweight process or decision, so interested to see what the specific advantages/disadvantages are there so we can call out and validate any assumptions being made here. 

Also, when there are a couple more updates, I'd like to get feedback from the community. This would benefit from being sent to kubeflow-discuss & having a spot for discussion in an upcoming Community Meeting.

How does that sound?

**Comment by richardsliu** [comment]
I would like to see a more detailed proposal for the migration plan. Specifically:

* How do we avoid having two divergent versions of the MpiJob?
* Assuming that the new MpiJob replaces the current version, how do we handle installation of the new MPI operator?
* How do we handle common dependencies?
* Will the new MpiJob API be backward compatible?
* What will be the release and versioning plan going forward?

**Review by Arhell — COMMENTED** [review]
/lgtm

**Comment by denkensk** [comment]
HI @ArangoGutierrez  Any new process？

**Comment by ArangoGutierrez** [comment]
Given https://github.com/cncf/toc/pull/950
let's
/hold

**Comment by ArangoGutierrez** [comment]
I think now I have time to get back to this thread
PING  @theadactyl @alculquicondor  @richardsliu 
What would it take to revive this thread

**Comment by alculquicondor** [comment]
While this is still open, it would be a lot of effort from a copyright perspective https://github.com/cncf/toc/pull/950

So I think we should wait.

**Comment by theadactyl** [comment]
I'd recommend waiting on CNCF donation review unless there is something
being significantly blocked here.

**Comment by tenzen-y** [comment]
cc

**Comment by ArangoGutierrez** [comment]
I guess we can restart this effort?

**Comment by ArangoGutierrez** [comment]
> I think we can rather drop it.
> 
> Now that kubeflow is part of CNCF, it's easier for organizations to contribute to the project.
> 
> @ArangoGutierrez do you have an additional motivation to make this a kubernetes subproject?

Agree
/Close

**Pull request closed without merging**